execution/aggregate/scalar_table.go

```diff
@@ -39,19 +40,23 @@ func newScalarTables(stepsBatch int, inputCache []uint64, outputCache []*model.S
 }
 
 func newScalarTable(inputSampleIDs []uint64, outputs []*model.Series, newAccumulator newAccumulatorFunc) *scalarTable {
-	accumulators := make([]*accumulator, len(outputs))
-	for i := 0; i < len(accumulators); i++ {
-		accumulators[i] = newAccumulator()
-	}
 	return &scalarTable{
-		inputs:       inputSampleIDs,
-		outputs:      outputs,
-		accumulators: accumulators,
+		inputs:         inputSampleIDs,
+		outputs:        outputs,
+		newAccumulator: newAccumulator,
 	}
 }
 
-func (t *scalarTable) aggregate(vector model.StepVector) {
-	t.reset()
+func (t *scalarTable) aggregate(arg float64, vector model.StepVector) {
+	if len(t.accumulators) == 0 {
```

Comment:
Why do we need to do this here? I thought we initialize tables in the Series call.

Reply:
Yeah.. i just commented about this: https://github.com/thanos-community/promql-engine/pull/116#discussion_r1017399624

The reason to initialize here was basically because the `arg` may not be calculated when calling `Series`.

But i wanna change this and put the argument on the `reset` method... And this make sense as the argument in theory value can change from one step (and so we should update it on every reset).

Ex: if we have a range query like `quantiles(scalar(min(something)),some_other_thing)`

Reply:
Ok got it. I think this makes sense 👍

Reply:
Updated :D